src/lib/gadgets/gadgets.unit-test.ts

```diff
@@ -45,3 +48,71 @@ equivalentAsync({ from: [maybeUint64], to: boolean }, { runs: 3 })(
     return await RangeCheck64.verify(proof);
   }
 );
+
+// --------------------------
+// ROT Gate
+// --------------------------
+let ROT = ZkProgram({
+  methods: {
+    run: {
+      privateInputs: [Field],
+      method(x) {
+        Gadgets.rot(x, 2, 'left');
+        Gadgets.rot(x, 2, 'right');
+      },
+    },
+  },
+});
+
+await ROT.compile();
+await equivalentAsync({ from: [maybeUint64], to: boolean }, { runs: 3 })(
+  (x) => {
+    if (x >= 1n << 64n) throw Error('expected 64 bits');
+    return true;
+  },
+  async (x) => {
+    let proof = await ROT.run(x);
+    return await ROT.verify(proof);
+  }
+);
+
+function testRot(
+  field: Field,
+  bits: number,
+  mode: 'left' | 'right',
+  result: Field
+) {
+  Provable.runAndCheck(() => {
+    let w = Provable.witness(Field, () => field);
+    let r = Provable.witness(Field, () => result);
+    let output = Gadgets.rot(w, bits, mode);
+    output.assertEquals(r, `rot(${field}, ${bits}, ${mode})`);
+  });
+}
+
+testRot(Field(0), 0, 'left', Field(0));
+testRot(Field(0), 32, 'right', Field(0));
+testRot(Field(1), 1, 'left', Field(2));
+testRot(Field(1), 63, 'left', Field('9223372036854775808'));
+testRot(Field(256), 4, 'right', Field(16));
+testRot(Field(1234567890), 32, 'right', Field(5302428712241725440));
+testRot(Field(2651214356120862720), 32, 'right', Field(617283945));
+testRot(Field(1153202983878524928), 32, 'right', Field(268500993));
```

Comment:
nit: these two values are larger than the max safe integer, and aren't captured exactly as JS numbers, i.e.
```
$ node
> 1153202983878524928
1153202983878525000
```

the test still works because it removes the lowest 32 bits with the right shift, and for that the values are easily close enough. I'd still use bigints

Reply:
Oh I just saw the expected result on the line above is also larger than the max